src/hotspot/cpu/riscv/vm_version_riscv.cpp

```diff
@@ -156,14 +156,35 @@ void VM_Version::initialize() {
     FLAG_SET_DEFAULT(UseSHA1Intrinsics, false);
   }
 
-  if (UseSHA256Intrinsics) {
-    warning("Intrinsics for SHA-224 and SHA-256 crypto hash functions not available on this CPU.");
-    FLAG_SET_DEFAULT(UseSHA256Intrinsics, false);
+  if (UseZvkn) {
```

Comment:
Maybe it's safe to move the code behind `#endif // COMPILER2` at line 291, as it depends on UseRVV.